src/games/strategy/triplea/ui/mapdata/RouteOptimizer.java

```diff
@@ -0,0 +1,185 @@
+          new Point2D.Double(point.getX() - mapWidth, point.getY() + mapHeight),
+          new Point2D.Double(point.getX() + mapWidth, point.getY() - mapHeight),
+          new Point2D.Double(point.getX() + mapWidth, point.getY() + mapHeight)));
+    }
+    if (isInfiniteX) {
+      result.addAll(Arrays.asList(
+          new Point2D.Double(point.getX() - mapWidth, point.getY()),
+          new Point2D.Double(point.getX() + mapWidth, point.getY())));
+
+    }
+    if (isInfiniteY) {
+      result.addAll(Arrays.asList(
+          new Point2D.Double(point.getX(), point.getY() - mapHeight),
+          new Point2D.Double(point.getX(), point.getY() + mapHeight)));
+
+    }
+    return result;
+  }
+
+  public static Point getPoint(Point2D point) {
+    return new Point((int) point.getX(), (int) point.getY());
+  }
+
+  public Point getLastEndPoint() {
+    return endPoint;
+  }
+
+  private List<Point[]> getAlternativePoints(Point... points) {
+    List<Point[]> alternativePoints = new ArrayList<>();
+    if (isInfiniteX || isInfiniteY) {
+      int altArrayCount = getAlternativePointArrayCount();
+      for (int i = 0; i < altArrayCount; i++) {
+        alternativePoints.add(new Point[points.length]);
+      }
+      int counter = 0;
+      for (Point point : points) {
+        Point normalizedPoint = normalizePoint(point);
+        if (isInfiniteX) {
+          alternativePoints.get(0)[counter] = new Point(normalizedPoint.x - mapWidth, normalizedPoint.y);
+          alternativePoints.get(1)[counter] = new Point(normalizedPoint.x + mapWidth, normalizedPoint.y);
+        }
+        if (isInfiniteY) {
+          int index = altArrayCount == 8 ? 2 : 0;
+          alternativePoints.get(index)[counter] = new Point(normalizedPoint.x, normalizedPoint.y - mapHeight);
+          alternativePoints.get(index + 1)[counter] = new Point(normalizedPoint.x, normalizedPoint.y + mapHeight);
+        }
+        if (isInfiniteX && isInfiniteY) {
+          alternativePoints.get(4)[counter] = new Point(normalizedPoint.x - mapWidth, normalizedPoint.y - mapHeight);
+          alternativePoints.get(5)[counter] = new Point(normalizedPoint.x - mapWidth, normalizedPoint.y + mapHeight);
+          alternativePoints.get(6)[counter] = new Point(normalizedPoint.x + mapWidth, normalizedPoint.y - mapHeight);
+          alternativePoints.get(7)[counter] = new Point(normalizedPoint.x + mapWidth, normalizedPoint.y + mapHeight);
+        }
+        counter++;
+      }
+    }
+    return alternativePoints;
+  }
+
+  public List<Point[]> getAllPoints(Point... points) {
+    List<Point[]> allPoints = getAlternativePoints(points);
+    Point[] normalizedPoints = new Point[points.length];
+    for (int i = 0; i < points.length; i++) {
+      normalizedPoints[i] = normalizePoint(points[i]);
+    }
+    allPoints.add(normalizedPoints);
+    return allPoints;
+  }
+
+  private int getAlternativePointArrayCount() {
+    if (isInfiniteX && isInfiniteY) {
+      return 8;
```

Comment:
magic numbers, please extract them to constants at the minimum to give them a name.